snuba/clickhouse/translators/snuba/allowed.py

```diff
@@ -50,7 +50,9 @@ class ColumnMapper(SnubaClickhouseMapper[Column, ValidColumnMappings]):
     pass
 
 
-class FunctionCallMapper(SnubaClickhouseMapper[FunctionCall, FunctionCall]):
+class FunctionCallMapper(
+    SnubaClickhouseMapper[FunctionCall, Union[FunctionCall, Literal]]
```

Comment:
Sorry this won't typecheck properly on translation.
Since the CurreidFunctionCall references a FunctionCall inside, a translator of a FunctionCall cannot resolve to NULL, otherwise translating a function can make the wrapping curriedfunction invalid, since translator only act locally.

There is some refactoring needed for this (the idea of fitting a function inside a curried function was not a good one).